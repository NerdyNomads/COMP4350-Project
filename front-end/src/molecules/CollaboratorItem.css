.CollaboratorItem {
  display: flex;
  flex-direction: row;
  width: calc(100%-5px);
  border-bottom: 1px solid #C6C6C6;
  padding: 5px;
}

.CollaboratorItem > .CollaboratorItem-remove {
  visibility: hidden;
}

.CollaboratorItem:hover > .CollaboratorItem-remove {
  visibility: visible;
}

.CollaboratorItem-remove {
  display: flex;
  margin-left: auto;
  cursor: pointer;
  user-select: none;
}
