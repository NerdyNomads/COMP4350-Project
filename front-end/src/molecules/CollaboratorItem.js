import React from "react";
import PropTypes from "prop-types";
import { TrashCanIcon } from "../atoms/icons";
import "./CollaboratorItem.css";

const CollaboratorItem = ({ text, remove, pending }) => {

  return (
    <div className="CollaboratorItem">
      <span className={`CollaboratorItem-email ${pending ? "CollaboratorItem-pending" : "" }`} >
        {pending ? text + "*" : text }
      </span>
      <TrashCanIcon className="CollaboratorItem-remove" onClick={remove}/>
    </div>
  );
};

CollaboratorItem.propTypes = {
  text: PropTypes.string.isRequired,
  pending: PropTypes.bool,
  remove: PropTypes.func.isRequired
};

export default CollaboratorItem;
