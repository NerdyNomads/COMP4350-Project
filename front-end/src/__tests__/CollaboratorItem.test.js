/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";

import CollaboratorItem from "../molecules/CollaboratorItem";

test("Should match the snapshot.", () => { 
  const mockFunc = () => null;
  let wrapper = shallow(<CollaboratorItem onRemove={mockFunc} email=""/>);
  expect(wrapper.html()).toMatchSnapshot();
});

test("Should show a '*' after the email if the user is still pending.", () => { 
  const mockFunc = () => null;
  let wrapper = shallow(<CollaboratorItem onRemove={mockFunc} email="test@email.com" pending={true}/>);

  let email = wrapper.find(".CollaboratorItem-pending");
  expect(email.length).toBe(1);
  expect(email.first().text()).toBe("test@email.com*");
});

test("Should show only the email (no '*') if the user is not pending.", () => { 
  const mockFunc = () => null;
  let wrapper = shallow(<CollaboratorItem onRemove={mockFunc} email="test@email.com" pending={false}/>);

  let email = wrapper.find(".CollaboratorItem-email");
  expect(email.length).toBe(1);
  expect(email.first().text()).toBe("test@email.com");
});

test("Should call the onRemove function if the trash can icon is clicked.", () => { 
  let onRemove = jest.fn();
  let wrapper = shallow(<CollaboratorItem onRemove={onRemove} email="test@email.com" pending={false}/>);

  let trashIcon = wrapper.find(".CollaboratorItem-remove").first();
  trashIcon.simulate("click");

  expect(onRemove).toBeCalledTimes(1);
});