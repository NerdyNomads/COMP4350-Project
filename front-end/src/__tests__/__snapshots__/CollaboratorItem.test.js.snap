// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot. 1`] = `"<div class=\\"CollaboratorItem\\"><span class=\\"CollaboratorItem-email \\"></span><img alt=\\"trash can\\" src=\\"/assets/images/trash_can.svg\\" class=\\"CollaboratorItem-remove\\"/></div>"`;
